### PR TITLE
[MISC] Dump model runner inputs when crashing

### PR DESCRIPTION
**Pull request by comaniac**
To better reproduce the model runner crashing due to illegal memory access and possibly other errors, this PR introduces a utility that dumps model runner inputs when crashing. Since the model runner inputs may be long, I dumped them using pickle for now. Any suggestions or better ideas are welcome.

cc @robertgshaw2-neuralmagic @simon-mo @DarkLight1337

### CODE REVIEW

File: vllm/worker/model_runner_base.py [hunk 1]
```diff
@@ -98,6 +101,37 @@ def _init_frozen_model_input_from_tensor_dict(
...
+
+    def _inner(func):
+
+        @wraps(func)
+        def _wrapper(*args, **kwargs):
+            try:
+                return func(*args, **kwargs)
+            except Exception as err:
+                timestamp = datetime.now().strftime("%Y%m%d-%H%M%S")
+                filename = f"err_{func.__name__}_input_{timestamp}.pkl"
```
Comment (simon-mo) [inline vllm/worker/model_runner_base.py:115]: maybe put it in tmp because some local file path are not writable?

**Comment by DarkLight1337** [comment]
Can we add instructions in the [GitHub issues template](https://github.com/vllm-project/vllm/blob/main/.github/ISSUE_TEMPLATE/400-bug%20report.yml) so users can share their logs upon encountering such errors?

**Comment by comaniac** [comment]
> Can we add instructions in the [GitHub issues template](https://github.com/vllm-project/vllm/blob/main/.github/ISSUE_TEMPLATE/400-bug%20report.yml) so users can share their logs upon encountering such errors?

Good point. Will do

**Comment by youkaichao** [comment]
do we need to add a flag for this? looks like some debugging feature that can also be added in https://docs.vllm.ai/en/latest/getting_started/debugging.html

**Comment by robertgshaw2-neuralmagic** [comment]
> do we need to add a flag for this? looks like some debugging feature that can also be added in https://docs.vllm.ai/en/latest/getting_started/debugging.html

The goal is to be able to get logs from production usage to help track down hard to replicate bugs (like illegal mem access in prefix caching). So having a flag defeats the purpose

**Comment by youkaichao** [comment]
makes sense then. please ignore my comment.

**Comment by comaniac** [comment]
@DarkLight1337 added to issue template. PTAL.

**Review by DarkLight1337 — APPROVED** [review]
LGTM. We should be careful when loading untrusted pickle files though.

**Comment by comaniac** [comment]
It should be fine as we never load it automatically? But yeah you may get virus if someone post a malicious pickle file to an issue...

**Pull request merged**